src/main/java/ui/components/NavigableListView.java

```diff
@@ -188,21 +188,17 @@ public void setOnItemSelected(IntConsumer callback) {
         onItemSelected = callback;
     }
 
-    public void selectFirstItem() {
-        requestFocus();
-        if (getItems().size() == 0) return;
-        getSelectionModel().clearAndSelect(0);
-        scrollAndShow(0);
-        selectedIndex = Optional.of(0);
-        onItemSelected.accept(selectedIndex.get());
-    }
+    /**
+     * Selects the nth item in the navigable list.
+     * @param n - Nth item, indexing is 1-based (1st item would correspond to n = 1)
+     */
 
-    public void selectLastItem() {
+    public void selectNthItem(int n) {
         requestFocus();
-        if (getItems().size() == 0) return;
-        getSelectionModel().clearAndSelect(getItems().size() - 1);
-        scrollAndShow(getItems().size() - 1);
-        selectedIndex = Optional.of(getItems().size() - 1);
+        assert(getItems().size() >= n);
```

Comment:
What I meant was `assert n > 0 && n <= size`. It's good to put `n` on the left because we read from left to right, and `n` is the variable whose range we are constraining. Also `assert` isn't a keyword so we don't need to parenthesise.